src/linter/ui5Types/utils/UniqueNameCreator.ts

```diff
@@ -51,6 +51,7 @@ const reservedJSLanguageKeywords = [
 	"while",
 	"with",
 	"yield",
+	"object",
```

Comment:
In what context is "object" a reserved JS language keyword?

Reply:
If the intention is to avoid covering the built-in `Object`, then we should rather implement this for any global name (e.g. using npm package [globals](https://www.npmjs.com/package/globals)).

But I wouldn't mix it with the reserved keyword check, it is on a different level. It's not syntactically forbidden, it just may have side effects by hiding built-in functionality.